pkg/secrets/check_rights_nix.go

```diff
@@ -13,15 +13,23 @@ import (
 	"syscall"
 )
 
-func checkRights(path string) error {
+func checkRights(path string, allowGroupExec bool) error {
 	var stat syscall.Stat_t
 	if err := syscall.Stat(path, &stat); err != nil {
 		return fmt.Errorf("invalid executable '%s': can't stat it: %s", path, err)
 	}
 
 	// checking that group and others don't have any rights
-	if stat.Mode&(syscall.S_IRWXG|syscall.S_IRWXO) != 0 {
-		return fmt.Errorf("invalid executable '%s', 'groups' or 'others' have rights on it", path)
+	if allowGroupExec {
```

Comment:
How would the user ID check below work with `allowGroupExec`?

Reply:
we don't remove owner permission. so the [check](https://github.com/DataDog/datadog-agent/pull/6298/files#diff-84a4058c436ae98c7c84d1b1bddee17cR36) should work

Reply:
I understand, but why would this allow a group to execute something even if it is group-executable but _not_ owned by the current user in that group? Again, not a security issue, but a usability one.

Reply:
it is to follow this openshift guideline about "Support arbitrary user ids": https://docs.openshift.com/container-platform/4.1/openshift_images/create-images.html#images-create-guide-openshift_create-images